### ISSUE TITLE
testnet panicked at 'called `Option::unwrap()` on a `None` value'

### ISSUE THREAD

**Issue by friedger** [blockstack/stacks-blockchain#1368]
testnet panicked at 'called `Option::unwrap()` on a `None` value' 
while I was deploying my smart contract oi-license (https://github.com/friedger/oi-license/blob/master/contracts/license/license.clar)
```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/vm/clarity.rs:277:9
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1052
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1426
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:204
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:224
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:472
  11: rust_begin_unwind
             at src/libstd/panicking.rs:380
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  13: core::panicking::panic
             at src/libcore/panicking.rs:52
  14: core::option::Option<T>::unwrap
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libcore/macros/mod.rs:10
  15: blockstack_lib::vm::clarity::ClarityBlockConnection::commit_to_block
             at src/vm/clarity.rs:277
  16: blockstack_lib::chainstate::stacks::db::ClarityTx::commit_to_block
             at src/chainstate/stacks/db/mod.rs:289
  17: blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::append_block
             at src/chainstate/stacks/db/blocks.rs:2487
  18: blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::process_next_staging_block
             at src/chainstate/stacks/db/blocks.rs:2648
  19: blockstack_lib::chainstate::stacks::db::blocks::<impl blockstack_lib::chainstate::stacks::db::StacksChainState>::process_blocks
             at src/chainstate/stacks/db/blocks.rs:2718
  20: blockstack_lib::testnet::node::Node::process_tenure
             at src/testnet/node.rs:310
  21: blockstack_lib::testnet::run_loop::RunLoop::start
             at src/testnet/run_loop.rs:185
  22: blockstack_core::main
             at src/main.rs:215
  23: std::rt::lang_start::{{closure}}
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  24: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  25: std::panicking::try::do_call
             at src/libstd/panicking.rs:305
  26: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:86
  27: std::panicking::try
             at src/libstd/panicking.rs:281
  28: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  29: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  30: std::rt::lang_start
             at /rustc/b8cedc00407a4c56a3bda1ed605c6fc166655447/src/libstd/rt.rs:67
  31: main
  32: __libc_start_main
  33: _start
```

**Comment by lgalabru** [comment]
Thank you for opening this issue @friedger.

Looking at your contract, your code looks clean, but unfortunately outdated: functions have been renamed and some of the syntax have changed.
We need to update our documentation website, meanwhile you can use this doc as a reference  https://github.com/blockstack/stacks-blockchain/blob/master/src/vm/docs/mod.rs.

Concerning the crash itself, this unwrap is crashing the testnet
https://github.com/blockstack/stacks-blockchain/blob/ebe4d3e6a6e9a32d97610ce833a12f125a3e2d53/src/vm/clarity.rs#L271-L278
@kantai do we have an action on our end?

**Comment by friedger** [comment]
@lacabra Thanks for the quick answer. Using `@blockstack/clarity` 0.1.13-alpha5, I can "deploy" the contract without any syntax errors. What is the best way to find the error?

**Comment by kantai** [comment]
> @kantai do we have an action on our end?

Yep -- that's a bug in the Clarity transaction code -- an error thrown by the "analyze contract" routine doesn't replace the cost tracker. Will open an issue and fix.

**Comment by kantai** [comment]
> Using @blockstack/clarity 0.1.13-alpha5, I can "deploy" the contract without any syntax errors. What is the best way to find the error?

You should build from `master` and use the Clarity CLI to analyze and deploy the contract -- it will provide more detailed syntax error messages.

**Comment by kantai** [comment]
Closing this issue since this bug was addressed via #1373
